python/lsst/jointcal/jointcal.py

```diff
@@ -248,7 +245,50 @@ def run(self, dataRefs, profile_jointcal=False):
         if associations.fittedStarListSize() == 0:
             raise RuntimeError('No stars in the fittedStarList!')
 
-        fit = jointcalLib.AstromFit(associations, spm, self.config.posError)
+        fit, astrom_model, sky2TP = self._fit_astrometry(associations)
+        fit, model = self._fit_photometry(associations)
+
+        # TODO: not clear that this is really needed any longer?
+        # TODO: MakeResTuple should at least be renamed, if we do want to keep that big data-dump around.
+        # Fill reference and measurement n-tuples for each tract
+        tupleName = "res_" + str(dataRefs[0].dataId["tract"]) + ".list"
+        fit.MakeResTuple(tupleName)
+
+        self._write_results(associations, astrom_model, dataRefs)
+
+        return pipeBase.Struct(dataRefs=dataRefs, oldWcsList=oldWcsList)
+
+    def _fit_photometry(self, associations):
+        """Fit the photometric data."""
+
+        print("====== Starting photometric fitting")
```

Comment:
Sorry to see all these print statements; I assume switching to logging is out of scope.